### ISSUE TITLE
Injector fails to start with `Init:ErrImageNeverPull`

### ISSUE THREAD

**Issue by TsarFox** [defenseunicorns/zarf#501]
### Environment
Device and OS: AWS EC2, Ubuntu 20.04
App version: v0.19.1
Kubernetes distro being used: v1.22.9+rke2r2 (d7c26a45b92cf3f76c063e93f8c6448fde7b2456)

### Steps to reproduce
1. Create an RKE2 cluster.
2. Run `zarf init`.

### Expected result
Command succeeds and Zarf is initialized in the cluster.

### Actual Result

```
  ✔  Gathering cluster information
      WARNING  Pod lookup timeout exceeded                                                                                                                          
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/defenseunicorns/zarf/src/internal/k8s.(*Tunnel).getAttachablePodForService(0xc000a60080)
        /home/runner/work/zarf/zarf/src/internal/k8s/tunnel.go:355 +0x17a
github.com/defenseunicorns/zarf/src/internal/k8s.(*Tunnel).getAttachablePodForResource(0xc000a60080)
        /home/runner/work/zarf/zarf/src/internal/k8s/tunnel.go:335 +0x95
github.com/defenseunicorns/zarf/src/internal/k8s.(*Tunnel).Establish(0xc000a60080)
        /home/runner/work/zarf/zarf/src/internal/k8s/tunnel.go:223 +0x245
github.com/defenseunicorns/zarf/src/internal/packager.hasSeedImages(0xc00000dce0)
        /home/runner/work/zarf/zarf/src/internal/packager/injector.go:203 +0x12f
github.com/defenseunicorns/zarf/src/internal/packager.runInjectionMadness({{0xc000672a20, 0x14}, {0xc000ae4510, 0x22}, {0xc000ae4540, 0x22}, {0xc000ae4570, 0x23}, {0xc000ada960, 0x1f}, ...})
        /home/runner/work/zarf/zarf/src/internal/packager/injector.go:96 +0x678
github.com/defenseunicorns/zarf/src/internal/packager.preSeedRegistry({{0xc000672a20, 0x14}, {0xc000ae4510, 0x22}, {0xc000ae4540, 0x22}, {0xc000ae4570, 0x23}, {0xc000ada960, 0x1f}, ...})
        /home/runner/work/zarf/zarf/src/internal/packager/seed.go:119 +0x758
github.com/defenseunicorns/zarf/src/internal/packager.deployComponents({{0xc000672a20, 0x14}, {0xc000ae4510, 0x22}, {0xc000ae4540, 0x22}, {0xc000ae4570, 0x23}, {0xc000ada960, 0x1f}, ...}, ...)
        /home/runner/work/zarf/zarf/src/internal/packager/deploy.go:223 +0x47d
github.com/defenseunicorns/zarf/src/internal/packager.Deploy()
        /home/runner/work/zarf/zarf/src/internal/packager/deploy.go:107 +0x1658
github.com/defenseunicorns/zarf/src/cmd.glob..func4(0x52d1ca0?, {0x3119e73?, 0x0?, 0x0?})
        /home/runner/work/zarf/zarf/src/cmd/initialize.go:28 +0x114
github.com/spf13/cobra.(*Command).execute(0x52d1ca0, {0x53567d0, 0x0, 0x0})
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:860 +0x663
github.com/spf13/cobra.(*Command).ExecuteC(0x52cfc20)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:974 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:902
github.com/defenseunicorns/zarf/src/cmd.Execute()
        /home/runner/work/zarf/zarf/src/cmd/root.go:45 +0x25
main.main()
        /home/runner/work/zarf/zarf/src/main.go:8 +0x17
```

### Visual Proof (screenshots, videos, text, etc)

![image](https://user-images.githubusercontent.com/6866190/171734119-764ab86a-62bf-4b17-9052-ee15af514669.png)

### Severity/Priority

:shrug: 

It's a blocker for me. I don't know how much you care about my use-case.

### Additional Context

Attempts to use `index.docker.io/rancher/hardened-etcd` if nothing else is installed. I was able to force Zarf to use `docker.io/ubuntu`, but this did not help.

**Comment by YrrepNoj** [comment]
I'll take a quick look at this tonight.

**Comment by jeff-mccoy** [comment]
Thanks that was super helpful.  I have a pretty good idea of what is happening now, could you also run 

`kubectl get pods --all-namespaces -o=jsonpath='{range .items[*]}{"\n"}{.metadata.name}{":\t"}{range .spec.containers[*]}{.image}{", "}{end}{end}' |\
sort`    

(source: https://kubernetes.io/docs/tasks/access-application-cluster/list-all-running-container-images/)

To get the list of pod -> image mappings.  This will confirm the issue so we can build a fix for it.  I'll explain what's happening in more detail once we confirm it is what I think it is.

**Comment by jeff-mccoy** [comment]
Thanks @TsarFox, working on some options now to fix this correctly.

**Comment by tworcester** [comment]
@jeff-mccoy I had the same issue on RKE, pulled the latest and did `make build-test`,  the above MR fixed the issue for me 👍

**Comment by jeff-mccoy** [comment]
Awesome, we're going to cut a release today for that, thanks!

**Comment by jeff-mccoy** [comment]
@TsarFox a fix is in [v0.19.4](https://github.com/defenseunicorns/zarf/releases/tag/v0.19.4)

**Comment by TsarFox** [comment]
Took me a bit to getting around and testing it, but it looks like 0.19.4 solves the issue. Thank you!
